pkg/autoscaler/metrics/collector.go

```diff
@@ -279,6 +279,10 @@ func newCollection(metric *av1alpha1.Metric, scraper StatsScraper, tickFactory f
 				scrapeTicker.Stop()
 				return
 			case <-scrapeTicker.C:
+				if c.metric.Spec.ScrapeTarget == "" {
+					// don't scrape empty target service
+					continue
```

Comment:
For other reviewers: Yes, it would be cooler if we stopped the actual goroutine and restarted it. But also vastly more complex. This is a net-benefit as is so I think this is good to leave like this :).